postgres-derive-test/src/composites.rs

```diff
@@ -238,3 +238,67 @@ fn raw_ident_field() {
 
     test_type(&mut conn, "inventory_item", &[(item, "ROW('foo')")]);
 }
+
+#[test]
+fn generics() {
+    #[derive(FromSql, Debug, PartialEq)]
+    struct InventoryItem<T: FromSqlOwned, U>
+    where
+        U: FromSqlOwned,
```

Comment:
Are these bounds actually required?

Reply:
`FromSqlOwned` is required for `#[derive(FromSql)]` which is needed to run the test. Or were you asking about the derived `Debug` and `PartialEq`?

Reply:
Derive logic should normally insert those bounds on the generated impls as necessary - you don't need `where T: Clone` at the struct definition when using `#[derive(Clone)]` for example.

Reply:
Just to make sure I understand the change request, the idea is add a FromSql<'a> bound on any generic type in the FromSql derive logic?

Reply:
Yep.

Reply:
So I started implementing this, and ran into an ambiguity (where this differs from `Clone`, which is in the rust prelude). I need to create a path to the FromSql and ToSql traits, however it could be one of a few options:
1. From the postgres_derive crate (`postgres_derive::FromSql<'a, T>`, but I think most uses will not import this crate directly)
2. From the postgres_types crate (`postgres_types::FromSql<'a, T>`)
3. Some other crate that re-exports or renames those types (no way to use a path from a crate name)
4. Assume that it's `use`d somewhere (`FromSql<'a, T>`)
Any suggestions on which assumption to use?

Reply:
Here's how the path is already produced in the existing logic: https://github.com/sfackler/rust-postgres/blob/master/postgres-derive/src/fromsql.rs#L90

Reply:
Thanks for pointing that out. Based on that, looks like it was option (2). Branch is updated, and tests pass locally.